src/verify_sig.rs

```diff
@@ -0,0 +1,58 @@
+use rsa::{RsaPrivateKey, RsaPublicKey};
+use rsa::pkcs1::{DecodeRsaPrivateKey, DecodeRsaPublicKey};
+use rsa::pkcs1v15::{SigningKey, VerifyingKey};
+use rsa::signature::{SignatureEncoding, Signer, Verifier};
+use rsa::sha2::Sha256;
+use std::{fs, str};
+
+// Takes a data buffer, a private key, and a public key, to sign the data
+// with the private key and verify the data with the public key.
+// Assume that we rely on RSA PKCS1v1.5, sha256 for the hash.
+pub fn verify_sig_rsa_pkcs1(databuf: &[u8], private_key: RsaPrivateKey, public_key: RsaPublicKey) {
+    // Equivalent of:
+    //   openssl rsautl -verify -pubin -key |public_key_path}
+    //   - in |sig_data| -out |out_hash_data|
+
+    let signing_key = SigningKey::<Sha256>::new(private_key);
+
+    // sign data with the key
+    let signature = signing_key.sign(databuf);
+    assert_ne!(signature.to_bytes().as_ref(), databuf);
+
+    // verify the signature
+    let verifying_key = VerifyingKey::<Sha256>::new(public_key);
+    verifying_key.verify(databuf, &signature).expect("failed to verify a key");
+}
+
+pub fn get_private_key_pkcs1_pem(private_key_path: &str) -> RsaPrivateKey {
```

Comment:
it's much more typical to use pkcs8 for pem files, that's also the default in `openssl genrsa`.

Reply:
Good question.
I tried to change the keys to RSA PKCS8 keys, but did not work. Still investigating.
Looks like the Rust RSA library does not fully support the RSA PKCS8 signature verification. Only PKCS1 works.

Reply:
Never mind.
Have managed to get pkcs8 working.
Naming scheme of the RSA lib is slightly inconsistent.